src/imitation/algorithms/adversarial.py

```diff
@@ -403,28 +384,39 @@ def _build_disc_feed_dict(
         )
 
         # Calculate generator-policy log probabilities.
-        log_act_prob = self._gen_policy.action_probability(obs, actions=acts, logp=True)
+        with th.no_grad():
+            obs_th = th.as_tensor(obs, device=self.gen_algo.device)
+            acts_th = th.as_tensor(acts, device=self.gen_algo.device)
+            _, log_act_prob_th, _ = self.gen_algo.policy.evaluate_actions(
+                obs_th, acts_th
+            )
+            log_act_prob = log_act_prob_th.detach().cpu().numpy()
+            del obs_th, acts_th, log_act_prob_th  # unneeded
         assert len(log_act_prob) == n_samples
         log_act_prob = log_act_prob.reshape((n_samples,))
 
-        fd = {
-            self.discrim.obs_ph: obs,
-            self.discrim.act_ph: acts,
-            self.discrim.next_obs_ph: next_obs,
-            self.discrim.done_ph: dones,
-            self.discrim.labels_gen_is_one_ph: labels_gen_is_one,
-            self.discrim.log_policy_act_prob_ph: log_act_prob,
+        batch_dict = {
+            "state": self._torchify_with_space(obs, self.discrim.observation_space),
+            "action": self._torchify_with_space(acts, self.discrim.action_space),
+            "next_state": self._torchify_with_space(
+                next_obs, self.discrim.observation_space
+            ),
+            "done": self._torchify_array(dones),
+            "labels_gen_is_one": self._torchify_array(labels_gen_is_one),
+            "log_policy_act_prob": self._torchify_array(log_act_prob),
         }
-        return fd
+
+        return batch_dict
 
 
 class GAIL(AdversarialTrainer):
     def __init__(
         self,
         venv: vec_env.VecEnv,
         expert_data: Union[types.Transitions, dataset.Dataset[types.Transitions]],
-        gen_policy: base_class.BaseRLModel,
+        gen_algo: base_class.BaseAlgorithm,
         *,
+        # FIXME(sam) pass in discrim net directly; don't ask for kwargs indirectly
         discrim_kwargs: Optional[Mapping] = None,
```

Comment:
Ugly; pass in fully-instantiated discriminator net instead.

Reply:
Leaving this for current PR. Will make the reward/discriminator interface more uniform in a future commit.